std/assembly/date.ts

```diff
@@ -11,155 +26,173 @@ export class Date {
     second: i32 = 0,
     millisecond: i32 = 0
   ): i64 {
-    return epochMillis(year, month + 1, day, hour, minute, second, millisecond);
+    if (year >= 0 && year <= 99) year += 1900;
+    var ms = epochMillis(year, month + 1, day, hour, minute, second, millisecond);
+    if (invalidDate(ms)) throw new RangeError(E_INVALIDDATE);
+    return ms;
   }
 
   @inline static now(): i64 {
     return <i64>Date_now();
   }
 
-  static fromString(dateTimeString: string): Date {
-    let hour: i32 = 0,
-        minute: i32 = 0,
-        second: i32 = 0,
-        millisecond: i32 = 0;
-    let dateString: string;
+  // It can parse only ISO 8601 inputs: YYYY-MM-DDThh:mm-ss.nnnnZ
+  @inline static parse(dateString: string): Date {
+    return this.fromString(dateString);
+  }
 
-    if (dateTimeString.includes("T")) {
+  static fromString(dateTimeString: string): Date {
```

Comment:
Should this be private, since it's not part of the Date spec?

Reply:
Yeah, it's not part of Date. But currently we can't use ad-hoc polymorphism for constructors so it's workaround solution for `new Date("2021-04-15")`.